### ISSUE TITLE
Issue with Regex in Goals

### ISSUE THREAD

**Issue by Birkaransingh94** [piwik/piwik#21164]
Hi Team,

There's an issue when a page visits goal has been setup with regex:
For E.g: 
Correct Regex is:
https:\/\/example\.example\.co\.uk\/checkout\/ABCD[0-9]+\/$

However, when saving it in goals, it removes "+" from the regex and saves it as:
**https:\/\/example\.example\.co\.uk\/checkout\/ABCD[0-9] \/$**

Thanks.

**Comment by michalkleiner** [comment]
Yeah, I can see how that would happen as `+` is a space in the URL format.

**Comment by michalkleiner** [comment]
Mildly related to #20875 where the validation would also help see this issue.

**Comment by sgiehl** [comment]
This looks like another problem related to https://github.com/matomo-org/matomo/issues/4231, but might already be possible to solve that by switching the goals api to use unsanitized parameters. See https://github.com/matomo-org/matomo/pull/19624

**Comment by michalkleiner** [comment]
@Birkaransingh94 can you see if Stefan's suggestion to switch the goals api to using unsanitised parameters helps?

**Comment by sgiehl** [comment]
@michalkleiner This is something we would need to do in the code

**Comment by michalkleiner** [comment]
Ah right, I thought that was already an existing option. Apologies for the confusion.

**Comment by ampaze** [comment]
Is someone working on this? As Matomo does not support Goal Tracking for past events, saving/creating an goal with .+ in the regex will lose valuable information.

**Comment by michalkleiner** [comment]
Not that I'm aware of at the moment, @ampaze. If you're keen to provide a PR we'll definitely look at it, otherwise it'll be up to the product team to prioritise the work on this.

**Comment by pmgsPT** [comment]
Fixed with https://github.com/matomo-org/matomo/pull/24691
